src/main/java/teammates/ui/controller/FeedbackSubmissionEditSaveAction.java

```diff
@@ -296,6 +298,16 @@ protected void checkAdditionalConstraints() {
         return;
     }
     
+    /**
+     * Note that when overriding this method, this should not use {@code respondingStudentList} 
+     * or {@code respondingInstructorList} of {@code FeedbackSessionAttributes}.
+     * 
+     * @return true if user has responses in the feedback session
+     */
+    protected boolean isUserRespondentOfSession() {
+        return hasValidResponse;
```

Comment:
Need to consider the case where not all questions are shown to the user, leading to cases where we can't only depend on the submitted form.
e.g. user is both student and instructor